crates/core/app/tests/mock_consensus.rs

```diff
@@ -80,54 +87,90 @@ async fn mock_consensus_can_send_a_spend_action() -> anyhow::Result<()> {
     let guard = common::set_tracing_subscriber();
     let storage = TempStorage::new().await?;
     let mut test_node = common::start_test_node(&storage).await?;
-    let mut rng = <rand_chacha::ChaChaRng as rand_core::SeedableRng>::seed_from_u64(0xBEEF);
 
     // Sync the mock client, using the test account's full viewing key, to the latest snapshot.
     let (viewing_key, spend_key) = (&test_keys::FULL_VIEWING_KEY, &test_keys::SPEND_KEY);
     let client = MockClient::new(viewing_key.deref().clone())
         .with_sync_to_storage(&storage)
-        .await?;
+        .await?
+        .tap(|c| info!(client.notes = %c.notes.len(), "mock client synced to test storage"));
 
     // Take one of the test account's notes...
     let (commitment, note) = client
         .notes
         .iter()
         .next()
         .ok_or_else(|| anyhow!("mock client had no note"))?
-        .tap(|(commitment, note)| {
-            tracing::info!(?commitment, ?note, "mock client note commitment")
-        });
+        .tap(|(commitment, note)| info!(?commitment, ?note, "mock client note commitment"));
 
-    // Build a transaction spending this note.
-    let tx = {
-        let position = client
+    let spend: ActionPlan = {
+        let proof = client
             .sct
-            .witness(*commitment)
-            .ok_or_else(|| anyhow!("commitment is not witnessed"))?
-            .position();
-        let spend = SpendPlan::new(&mut rng, note.clone(), position);
+            .witness(commitment.clone())
+            .ok_or_else(|| anyhow!("commitment is not witnessed"))?;
+        let note = note.clone();
+        let position = proof.position();
+        SpendPlan::new(&mut OsRng, note, position).into()
+    };
+
+    // Build a transaction spending this note.
+    let tx: Transaction = {
```

Comment:
a passing thought, but this part of the API is pretty long-winded. we talked a bit about this in our sprint planning meeting today, but i'd love to see a more convenient way to go about this for future test authors.

Reply:
The API we'd like to use is the `Planner` in the view crate. We'd need to be able to pass the `MockClient` to its plan() method: https://rustdoc.penumbra.zone/main/penumbra_view/struct.Planner.html#method.plan

Reply:
x-ref #3896, i think?

Reply:
Clarification post 54c0b35d8 , where the transaction plan creation becomes much simpler: my point is that for simple tests, where we can write down the complete transaction plan explicitly, we should do so. In this case, we can, which is what 54c0b35d8 does.

 If we need to do more complex test logic, test logic that is complex enough that we cannot simply write down the plan, then we should decide on a way to use the `Planner` API to plan transactions. This may involve pulling in the full rust view server implementation, or continuing to use the `MockClient`. But for simple cases it's important to be able to test the chain logic without all the complexity of the rust view server.